From baebac8a23d270ebac7f27a39a61e815a33021da Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:40:12 -0400
Subject: [virt] kvm/x86: preserve the high 32-bits of the PAT register

Message-id: <1406040016-3289-218-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86296
O-Subject: [RHEL7 PATCH v2 217/221] KVM: x86: preserve the high 32-bits of the PAT register
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

KVM does not really do much with the PAT, so this went unnoticed for a
long time.  It is exposed however if you try to do rdmsr on the PAT
register.

Reported-by: Valentine Sinitsyn <valentine.sinitsyn@gmail.com>
Cc: stable@vger.kernel.org
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 7cb060a91c0efc5ff94f83c6df3ed705e143cdb9)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/include/asm/kvm_host.h b/arch/x86/include/asm/kvm_host.h
index 4931415..9f83c17 100644
--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -461,7 +461,7 @@ struct kvm_vcpu_arch {
  bool nmi_injected;    /* Trying to inject an NMI this entry */
 
  struct mtrr_state_type mtrr_state;
- u32 pat;
+ u64 pat;
 
  unsigned switch_db_regs;
  unsigned long db[KVM_NR_DB_REGS];
-- 
1.7.1